requirements/dev.txt

```diff
@@ -1,4 +1,12 @@
--r tools.txt
+click>=7.0
+google-api-python-client>=1.8.2
+google_auth_oauthlib>=0.4.2
+httplib2>=0.10
+lxml>=3.0
+oauth2client>=4.0
+ply>=3.4
+python-dateutil>=2.6.0
+requests>=2.0
 pylint>=2.6.0
```

Comment:
Why? The point of this is to do not to have to keep the dependencies in sync in two different files. I suspect that Bazel does not support the `requirements.txt` file format, but I don't think this is a thing that we should work around here.

Reply:
I haven't found a way to pass tools.txt to `compile_pip_requirements` sandbox. [See here][cpr]. The `extra_deps` parameter can not be it, as it requires a `PyPi` provider, which a source file of course doesn't have

[cpr]: https://github.com/bazelbuild/rules_python/blob/main/docs/pip.md#compile_pip_requirements

Even though the underlying `pip` invocation will look for it, there currently does not seem to be a way to add that file to the sandbox where `compile_pip_requirements` is executing. This means that the rule will fail when executed with `File not found: requirements/tools.txt`.  

Since in my view not having a hermetic build is a bigger issue than some duplication, I opted to go this route.

Reply:
Also, why do you install the development dependencies?

Reply:
> Using bazel for deploying Python applications is an extremely niche use case

I suppose it is, and yet beancount has bazel support right there. Perhaps you want to add a `CONTRIBUTING.md`  to clarify your expectations?

Else, if it is a niche use case, why even use bazel?

> I'm not very sympathetic to the idea of making the life of the Beancount maintainers harder to satisfy this use case.

Sorry, there was only so much I could do in an evening after work.

> Bazel is developed by an organization with very deep pockets that employs a huge number of engineers. Ask them to fix their tools.

I filed a bug and will look into this when I have the time. Is there more I could be doing here?

Reply:
> Also, why do you install the development dependencies?

My goal here was to just rewrite what was already there in starlark. If there is a different way one should set this up, I welcome a pointer. A difference between `dev.txt` and `tools.txt` is not obvious to me.

Reply:
OK, I learned how to remove the duplication in dev.txt vs tools.txt, and the original issue in this thread is now fixed.